### ISSUE TITLE
LightGBM returns feture importance of 0 for a categorical feature with five categories (dataset: 130 instances)

### ISSUE THREAD

**Issue by hengjieyu** [microsoft/LightGBM#4491]
## Description

Using LightGBM for regression, our dataset has a  categorical feature (5 categories), totally 130 instances. The feature_importances_ returns 0 for the categorical feature. We used a public dataset to do same thing, and this issue happened again.

## Reproducible example

import pandas as pd
import numpy as np
import lightgbm as lgb
from sklearn.datasets import fetch_california_housing
from sklearn.model_selection import train_test_split
from sklearn.datasets import fetch_california_housing

cal_housing = fetch_california_housing()
X = pd.DataFrame(cal_housing.data, columns=cal_housing.feature_names)
y = cal_housing.target
y -= y.mean()

L = np.array([0,1,2,3,4])
LL = np.repeat(L,[30,30,20,20,30],axis=0)
X = X.iloc[0:130,:]
X["add"] = LL
X_train, X_test, y_train, y_test = train_test_split(X,y[0:130], test_size=0.1, random_state=0)
model = lgb.LGBMRegressor(n_jobs=-1,n_estimators=1000)
model.fit(X_train, y_train, categorical_feature=['add'])
LightGBM_importance = model.feature_importances_
LightGBM_importance

>>array([931, 208, 631, 629, 571, 372, 164, 194,   0])

## Environment info

LightGBM version or commit hash:
LightGBM version: 3.2.1.99 and 2.3.0, we all tried, and same issue happened.

Command(s) you used to install LightGBM

```shell

```

## Additional Comments

Decresing the category number to 3 or adding the number of instances to 450,  an feture importance greater than 0 is returned.  The code seems to be ok.

**Comment by hengjieyu** [comment]
It seems like LightGBM relys on the ratio of category number and instance number when treating a categorical feature. If the ratio is too big (5/130 in this example), the LightGBM model will ignore this categorical feature and return a feature importance of 0. When the ratio is 5/450 or 3/130 in this example, the model will use this categorical feature and return a feature importance greater than 0.
